Feat: add E2E tests for backfill pause and stop controls

KyeongJooni opened the pull request:
Add UI E2E tests for backfill pause, resume, and cancel controls

This PR adds end-to-end tests to verify the backfill control functionality on the DAG detail page.

**What this tests:**
- Pausing a running backfill and verifying status changes to paused
- Resuming a paused backfill and verifying status returns to running
- Cancelling an active backfill and verifying status changes to cancelled
- Verifying that cancelled backfills cannot be resumed

**Implementation:**
- Created `BackfillPage` page object following existing POM pattern
- Added test suite `backfill-controls.spec.ts` with 3 test cases
- All tests follow the same structure as existing E2E tests (`dags-list.spec.ts`)

closes: #59594

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

<!-- Please keep an empty line above the dashes. -->

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

KyeongJooni committed:
feat: add E2E tests for backfill pause and stop controls

Add Playwright E2E tests to verify pause, resume, and cancel
functionality for backfills on the DAG detail page.

- Create BackfillPage page object for backfill interactions
- Add tests for pause/resume workflow
- Add tests for cancel workflow
- Verify cancelled backfills cannot be resumed

Closes #59594

KyeongJooni committed:
feat: add E2E tests for backfill pause and stop controls

Add Playwright E2E tests to verify pause, resume, and cancel
functionality for backfills on the DAG detail page.

- Create BackfillPage page object for backfill interactions
- Add tests for pause/resume workflow
- Add tests for cancel workflow
- Verify cancelled backfills cannot be resumed

Closes #59594

choo121600 reviewed (commented):
Setting up prek and committing the changes should resolve the static check ci issue :)

https://github.com/apache/airflow/blob/main/contributing-docs/08_static_code_checks.rst

KyeongJooni committed:
fix: reorder methods alphabetically in BackfillPage

KyeongJooni committed:
fix: reorder methods alphabetically in BackfillPage

KyeongJooni committed:
refactor: align with PR #59692 authentication pattern

Remove redundant login calls from tests as authentication
is now handled globally via global-setup.ts

KyeongJooni committed:
refactor: align with PR #59692 authentication pattern

Remove redundant login calls from tests as authentication
is now handled globally via global-setup.ts

KyeongJooni committed:
Merge branch 'main' into e2e/backfill-pause-stop

choo121600 commented on airflow-core/src/airflow/ui/tests/e2e/pages/BackfillPage.ts:
```diff
@@ -0,0 +1,103 @@
+
+    await this.page.waitForTimeout(1000);
+
+    const backfillRadio = this.page.locator('input[value="backfill"]');
```
```suggestion
    const backfillRadio = this.page.locator('label:has-text("Backfill")');
```

choo121600 reviewed (commented):
Maybe you should be able to find the backfill radio button like this.

KyeongJooni committed:
Update airflow-core/src/airflow/ui/tests/e2e/pages/BackfillPage.ts

Co-authored-by: Yeonguk Choo <choo121600@gmail.com>

KyeongJooni committed:
fix: update locators to match existing patterns

- Update trigger button locator to match DagsPage
- Apply reviewer suggestion for backfill radio locator
